docs/contributors/code/native-mobile-integration-test-guide.md

```diff
@@ -0,0 +1,368 @@
+```
+
+```
+const radiusSlider = getByTestId( 'Slider Border Radius' );
+```
+
+Note that either a plain string or a regular expression can be passed into these queries. A regular expression is best for querying part of a string (e.g. any element whose accessibility label contains `Unsupported Block. Row 1`). Note that special characters such as `.` need to be escaped.
+
+## Use of `waitFor`
+
+After rendering the components or firing an event, side effects might happen due to potential state updates so the element we’re looking for might not be yet rendered. In this case, we would need to wait for the element to be available and for this purpose, we can use the `waitFor` function, which periodically executes the provided callback to determine whether the element appeared or not.
+
+Here are some examples:
+
+```
+const mediaLibraryButton = await waitFor( () =>
+	getByText( 'WordPress Media Library' )
+);
+```
+
+```
+const missingBlock = await waitFor( () =>
+	getByA11yLabel( /Unsupported Block\. Row 1/ )
+);
+```
+
+```
+const radiusSlider = await waitFor( () =>
+	getByTestId( 'Slider Border Radius' )
+);
+```
+
+In most cases we’ll use the `waitFor` function, but it’s important to note that it should be restricted to those queries that actually require waiting for the element to be available.
+
+NOTE: The `react-native-testing-library` package provides the `query*` and `find*` functions for this purpose too, but we should avoid using them for now because there’s a [known issue](https://github.com/callstack/react-native-testing-library/issues/379) that would make the test fail.
+
+## `within` queries
+
+It’s also possible to query elements contained in other elements via the `within` function, here is an example:
+
+```
+const missingBlock = await waitFor( () =>
+	getByA11yLabel( /Unsupported Block\. Row 1/ )
+);
+const translatedTableTitle = within( missingBlock ).getByText(
+	'Tabla'
+);
+```
+
+## Fire events
+
+As important as querying an element is to trigger events to simulate the user interaction, for this purpose we can use the `fireEvent` function ([documentation](https://callstack.github.io/react-native-testing-library/docs/api#fireevent)).
+
+Here is an example of a press event:
+
+**Press event:**
+
+```
+fireEvent.press( settingsButton );
+```
+
+We can also trigger any type of event, including custom events, in the following example you can see how we trigger the `onValueChange` event ([code reference](https://github.com/WordPress/gutenberg/blob/520cbd9d2af4bbc275d388edf92a6cadb685de56/packages/components/src/mobile/bottom-sheet/range-cell.native.js#L227)) for the Slider component:
+
+**Custom event – onValueChange:**
+
+```
+fireEvent( heightSlider, 'valueChange', '50' );
+```
+
+## Expect correct element behaviour
+
+After querying elements and firing events, we have to verify that the logic works as expected, for this purpose we can use the same expect function from Jest as we use in unit tests.
```

Comment:
The "expect" word has inline code format in the original document.
```suggestion
After querying elements and firing events, we have to verify that the logic works as expected, for this purpose we can use the same `expect` function from Jest as we use in unit tests.
```

Reply:
Thanks for catching and fixing that 🙇